src/createAsset.ts

```diff
@@ -51,25 +55,59 @@ export const getDataFromDBStream = async (stream: NodeJS.ReadableStream, schema:
 	return data;
 };
 
-export const keyString = (key: Buffer): string => key.toString('binary');
-
-export const getTransactions = async (blockID: Buffer, db: KVStore): Promise<Buffer[]> => {
-	const txIDs: Buffer[] = [];
-	const ids = await db.get(`${DB_KEY_TRANSACTIONS_BLOCK_ID}:${keyString(blockID)}`);
-	const idLength = 32;
-	for (let i = 0; i < ids.length; i += idLength) {
-		txIDs.push(ids.slice(i, i + idLength));
+export const getBlockHeadersByIDs = async (
+	db: KVStore,
+	arrayOfBlockIds: ReadonlyArray<Buffer>,
+): Promise<Buffer[]> => {
+	const blocks = [];
+	for (const id of arrayOfBlockIds) {
+		const block = await db.get(`${DB_KEY_BLOCKS_ID}:${keyString(id)}`);
+		blocks.push(block);
 	}
-	if (txIDs.length === 0) {
+	return blocks;
+};
+
+export const getBlocksIDsFromDBStream = async (stream: NodeJS.ReadableStream) => {
+	const blockIDs = await new Promise<Buffer[]>((resolve, reject) => {
+		const ids: Buffer[] = [];
+		stream
+			.on('data', ({ value }: { value: Buffer }) => {
+				ids.push(value);
+			})
+			.on('error', error => {
+				reject(error);
+			})
+			.on('end', () => {
+				resolve(ids);
+			});
+	});
+	return blockIDs;
+};
+
+export const getTransactions = async (blockID: Buffer, db: KVStore) => {
+	try {
+		const txIDs: Buffer[] = [];
+		const ids = await db.get(`${DB_KEY_TRANSACTIONS_BLOCK_ID}:${keyString(blockID)}`);
+		const idLength = 32;
+		for (let i = 0; i < ids.length; i += idLength) {
+			txIDs.push(ids.slice(i, i + idLength));
```

Comment:
Where `TRANSACTION_ID_LENGTH = SHA_256_HASH_LENGTH` in `src/constant.ts`.
```suggestion
		const txIDLength = TRANSACTION_ID_LENGTH;
		for (let i = 0; i < ids.length; i += txIDLength) {
			txIDs.push(ids.slice(i, i + txIDLength));
```